cmake/AutoRevision.cmake

```diff
@@ -0,0 +1,62 @@
+# This file runs a collection of git commands to help automate the process
+# of generating the build revision. It takes an update to 3 files by hand
+# to do it manually and automates it.
+
+# Auto Revision needs git to work
+find_package(Git)
+
+if(Git_FOUND)
+
+  # get the value of the git hash at HEAD to 5 chars
+  execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --short=5 HEAD
+                  OUTPUT_VARIABLE FC21_REV_HEAD_HASH_H
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
What if git is found, but this is not a git repo (eg building from a tarball)?

Reply:
Oh that is a good point. I need another if() layer.